db/sequence_allocator.go

```diff
@@ -381,3 +415,69 @@ func (s *sequenceAllocator) waitForReleasedSequences(ctx context.Context, startT
 	time.Sleep(requiredWait)
 	return requiredWait
 }
+
+// _fixSyncSeqRollback will correct a rolled back _sync:seq document in the bucket
+func (s *sequenceAllocator) _fixSyncSeqRollback(ctx context.Context, incrValue, expectedValue uint64) (allocatedToSeq uint64, err error) {
+	// grab _sync:seq value + its current cas value
+	var result uint64
+	cas, err := s.datastore.Get(s.metaKeys.SyncSeqKey(), &result)
```

Comment:
It looks like the retry flow here is:
1. Get _sync:seq value + cas
2. If value < targetValue, WriteCas to update
  2.1. On cas failure, return to 1
3. On success, incr(batchsize)

Can we move initial Get inside the worker to simplify this function, and to ensure we only do 3 in one place?